src/API.jl

```diff
@@ -1469,11 +1481,13 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
                         end
                         loaded && (n_loaded += 1)
                     catch err
-                        close(iob)
+                        # close pipes to end the std output monitor
+                        close(stderr_pipe)
+                        close(stdout_pipe)
                         wait(t_monitor)
```

Comment:
> There also should not be a close(iob) call, ever, since that corrupts the stream.

@vtjnash the closes here are to stop the async `monitor_stderr_stdout`. Is that acceptable?